### PR TITLE
feat(ST-112): add edit resource function

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added Edit Resource Modal component
- Added updateResource query in resources.api.ts
- Added a hook for calling api

## Associated Ticket Link(s)

- https://trello.com/c/Trfrno4t

## Related Pull Request(s)

- N/A

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/39a76be5-e010-42e6-8963-8b31bf9b105d

### CODE REVIEW

**Comment by ragib-hossain-sazim** [comment]
Should we load the uploaded document in the  edit modal?

File: modules/Classroom/components/EditResourceModal/EditResourceModal.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            )}
+          />
+
+          <Flex justify={"end"} align={"center"} gap={12} mb={10}>
+            <Button loading={isLoading} bg={"green"} onClick={() => reset()}>
+              Cancel
+            </Button>
+            <Button bg={"green"} type="submit">
+              Update
+            </Button>
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/components/EditResourceModal/EditResourceModal.tsx:104]: Please make the loading indicator and disable button changes as per discussion during demo

File: modules/Classroom/hooks/useEditResourceForm.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const onSubmit = async (data: TEditResourceFormData, resourceId: number) => {
+    const editResourceInfo: TEditResourceInfoDto = {
+      title: data?.title,
+      description: data?.description,
+    };
+
+    if (!resourceId) {
+      console.error("No resource found");
+      return;
+    }
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/hooks/useEditResourceForm.ts:22]: I am not sure if you need this check

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left some comments

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**